src/book.py

```diff
@@ -188,36 +188,95 @@ def _read_coinbase(self, file_path: Path) -> None:
 
-                self.append_operation(
-                    operation, utc_time, platform, change, coin, row, file_path
-                )
+                if operation == "Convert":
+                    # parse change + coin from remark which is in format "0,123 ETH to 0,456 BTC"
+                    remark_target = remark.split(' to ')[-1]
+                    remark_target_parts = remark_target.split(' ')
+                    _convert_change = remark_target_parts[0].replace(',', '.')
+                    convert_change = misc.xdecimal(_convert_change)
+                    convert_coin = remark_target_parts[1]
+
+                    self.append_operation("Sell", utc_time, platform,
+                                          change, coin, row, file_path)
+                    self.append_operation("Buy", utc_time, platform,
+                                          convert_change, convert_coin, row, file_path)
+                else:
+                    self.append_operation(
+                        operation, utc_time, platform, change, coin, row, file_path
+                    )
+
+                    if operation == "Sell":
+                        assert isinstance(eur_subtotal, decimal.Decimal)
+                        self.append_operation(
+                            "Buy", utc_time, platform, eur_subtotal, "EUR", row, file_path
+                        )
+                    elif operation == "Buy":
+                        assert isinstance(eur_subtotal, decimal.Decimal)
+                        self.append_operation(
+                            "Sell", utc_time, platform, eur_subtotal, "EUR", row, file_path
+                        )
+
+                    if eur_fee:
+                        self.append_operation(
+                            "Fee", utc_time, platform, eur_fee, "EUR", row, file_path
+                        )
 
-                # Save price in our local database for later.
-                self.price_data.set_price_db(platform, coin, "EUR", utc_time, eur_spot)
+    def _read_coinbase_pro(self, file_path: Path) -> None:
+        platform = "coinbase_pro"
+        operation_mapping = {
+            "BUY": "Buy",
+            "SELL": "Sell",
+        }
+
+        with open(file_path, encoding="utf8") as f:
+            reader = csv.reader(f)
+
+            # Skip header.
+            next(reader)
+
+            for portfolio, trade_id, product, operation, _utc_time, _size, size_unit, _price, _fee, total, price_fee_total_unit in reader:
+                row = reader.line_num
+
+                # Parse data.
+                utc_time = datetime.datetime.strptime(
+                    _utc_time, "%Y-%m-%dT%H:%M:%S.%fZ")
+                utc_time = utc_time.replace(tzinfo=datetime.timezone.utc)
+                operation = operation_mapping.get(operation, operation)
+                size = misc.xdecimal(_size)
+                price = misc.xdecimal(_price)
+                fee = misc.xdecimal(_fee)
```

Comment:
Consider using `misc.force_decimal`, when the value is supposed to be a `Decimal`.

Reply:
Ah, I see.. I believe you are not using `mypy`. Pulling the rebasing to origin/main should add your branch our new linter workflow which helps you to detect errors like this and more.

For easy local usage, I created a `Makefile` with the relevant commands. If you have not `make` installed, you can copy/paste them into the terminal.

Reply:
This should probably be fine since now. I've just configured VS Code properly:

```json
{
    "python.linting.enabled": true,
    "python.linting.flake8Enabled": true,
    "python.linting.mypyEnabled": true
}
```